### PR TITLE
PP-6007: Card details page validation test

### PR DESCRIPTION
**Pull request by oswaldquek**
## WHAT
_A brief description of the pull request:_

## HOW 
_Steps to test or reproduce:_

### CODE REVIEW

File: test/cypress/integration/card/card-details-page-validation.spec.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+const checkCardDetailsStubs = cardPaymentStubs.checkCardDetailsStubs(chargeId)
+
+describe('Card details page validation', () => {
+  beforeEach(() => {
+    // this test is for the full process, the session should be maintained
+    // as it would for an actual payment flow
+    Cypress.Cookies.preserveOnce('frontend_state')
+  })
+
+  it('Should setup the payment and load the page', () => {
```
Comment (kbottla) [inline test/cypress/integration/card/card-details-page-validation.spec.js]: Is this required and intentional ? There is a test already here https://github.com/alphagov/pay-frontend/blob/master/test/cypress/integration/card/payment.spec.js#L77

File: test/cypress/integration/card/card-details-page-validation.spec.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    cy.task('setupStubs', createPaymentChargeStubs)
+    cy.visit(`/secure/${tokenId}`)
+
+    cy.location('pathname').should('eq', `/card_details/${chargeId}`)
+    cy.window().its('chargeId').should('eq', `${chargeId}`)
+  })
+
+  it('Should enter card details', () => {
```
Comment (kbottla) [inline test/cypress/integration/card/card-details-page-validation.spec.js]: Same as above

File: test/cypress/integration/card/card-details-page-validation.spec.js (hunk 1 above)
Comment (oswaldquek) [reply test/cypress/integration/card/card-details-page-validation.spec.js]: we need this setup for this test to get to the card details page so we can test the errors

File: test/cypress/integration/card/card-details-page-validation.spec.js (hunk 2 above)
Comment (oswaldquek) [reply test/cypress/integration/card/card-details-page-validation.spec.js]: we need this setup for this test to get to the card details page so we can test the errors

File: test/cypress/integration/card/card-details-page-validation.spec.js (hunk 1 above)
Comment (kbottla) [reply test/cypress/integration/card/card-details-page-validation.spec.js]: Can these steps be not part of `Submitting confirmation should show errors` test itself and add the test to `payment.spec.js` ? Not sure of the convention followed for cypress tests

```
cy.task('setupStubs', createPaymentChargeStubs)
cy.visit(`/secure/${tokenId}`)
cy.server()
cy.route('POST', `/check_card/${chargeId}`)
```

File: test/cypress/integration/card/card-details-page-validation.spec.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    cy.task('setupStubs', checkCardDetailsStubs)
+    cy.server()
+    cy.route('POST', `/check_card/${chargeId}`).as('checkCard')
```
Comment (kbottla) [inline test/cypress/integration/card/card-details-page-validation.spec.js]: (Minor) Don't need alias `as('checkCard')` as it is not used

File: test/cypress/integration/card/card-details-page-validation.spec.js (hunk 1 above)
Comment (oswaldquek) [reply test/cypress/integration/card/card-details-page-validation.spec.js]: `payment.spec.js` tests for card number validation. I didn't want to put new validation tests there as there spec is already rather large. I think a better option would be to move the card number validation test from `payment.spec.js` to this new spec.

**Pull request merged**